cpp-package/example/inference/simple_rnn.cpp

```diff
@@ -0,0 +1,372 @@
+      return index;
+    } else {
+      index++;
+    }
+  }
+  throw std::runtime_error("The output symbol name can not be found");
+}
+
+
+/*
+ * The following function runs the forward pass on the model.
+ * The executor is created in the constructor.
+ */
+std::string Predictor::PredictText(const std::string& input_text) {
+  /*
+   * Initialize a vector of length equal to 'sequence_length' with 0.
+   * Convert the input string to a vector of indices that represent
+   * the words in the input string.
+   */
+  std::vector<float> index_vector(sequence_length, 0);
+  ConverToIndexVector(input_text, &index_vector);
+
+  executor->arg_dict()["data"].SyncCopyFromCPU(index_vector.data(), sequence_length);
+
+  // Run the forward pass.
+  executor->Forward(false);
+
+  /*
+   * The output is available in executor->outputs. It is a vector of
+   * NDArray. We need to find the index in that vector that
+   * corresponds to the output symbol "softmax_layer_output".
+   * The output "softmax_layer_output" has shape [sequence_length, vocab_size]
+   * The vocab_size in this case is equal to the size of dictionary.
+   */
+  const std::string output_symbol_name = "softmax_layer_output";
+  int output_index = GetIndexForOutputSymbolName(output_symbol_name);
+  std::vector<NDArray> outputs = executor->outputs;
+  auto arrayout = executor->outputs[output_index].Copy(global_ctx);
+
+  /*
+   * The output will contain the probability distribution for each of the
+   * word in sequence.
+   * We will run ArgmaxChannel operator to find out the index with the
+   * highest probability. This index will point to word in the predicted
+   * output string.
+   */
+  arrayout = arrayout.ArgmaxChannel();
+  arrayout.WaitToRead();
+
+  std::ostringstream oss;
+  for (std::size_t i = 0; i < static_cast<size_t>(sequence_length); ++i) {
+    auto charIndex = arrayout.At(0, i);
+    oss << intToWord[charIndex] << " ";
+  }
+  return oss.str();
+}
+
+
+Predictor::~Predictor() {
+  if (executor) {
+    delete executor;
+  }
+  MXNotifyShutdown();
+}
+
+
+void printUsage() {
+    std::cout << "Usage:" << std::endl;
+    std::cout << "simple_rnn " << std::endl
+              << "[--input] Input string sequence. "
+              << "e.g. \"Good morning. I appreciate the opportunity to speak here\""  << std::endl
+              << "[--gpu]  Specify this option if workflow needs to be run in gpu context "
+              << std::endl;
+}
+
+
+void Download_files(const std::vector<std::string> model_files) {
+  std::string wget_command = "wget -nc ";
+  std::string s3_url = "https://s3.amazonaws.com/mxnet-cpp/RNN_model/";
+  for (auto &file : model_files) {
+    std::ostringstream oss;
+    oss << wget_command << s3_url << file << " -O " << file;
+    int status = system(oss.str().c_str());
+    LG << "Downloading " << file << " with status " << status;
+  }
+  return;
+}
+
+
+int main(int argc, char** argv) {
+  std::string model_file_json = "./obama-speaks-symbol.json";
```

Comment:
Should the input file paths be hardcoded here ? 
Can they be passed as parameters to the main method in ```argv``` ?

Reply:
My bad, saw later that these are passed to the S3 Bucket to download.